### DISCUSSION TITLE
Surface field schema errors more gracefully

### DISCUSSION THREAD

**Discussion by Mng-dev-ai** [payloadcms/payload#2906] [Roadmap]
### Link to reproduction

n/a

### To Reproduce

Create a collection with the following code:
```
export function Virtual(props) {
    const Virtual = {
        type: 'json',
        ...props,
        admin: {
            hidden: true,
            ...props.admin,
        },
        hooks: {
            beforeChange: [
                ({ siblingData }) => {
                    siblingData[props.name] = undefined;
                },
            ],
            afterRead: [props.computed],
        },
    };
    return Virtual;
}

const Sandbox = {
    slug: 'sandbox',
    fields: [
        Virtual({
            name: 'virtual',
            computed: (data) => {
                return {}; // some logic
            },
        }),
    ],
};

export default Sandbox;

```

### Describe the Bug

When the server reloads you'll see the following error:

`Collection "sandbox" > Field "virtual" > "value" does not match any of the allowed types`

but the actual error is about `computed` not being allowed in a field definition

### Payload Version

1.9.2

**Comment by ssyberg** [comment]
Have frequently experienced this when doing dev and not being vigilant about watching hot reloads, only workaround I've found is commenting out all fields and reintroducing them one by one back into the collection.

**Comment by jmikrut** [comment]
Hey all, this is actually just how our implementation of Joi schema validation works. Nested errors are fairly difficult to report properly. 

As of now, this is technically functioning as we intended, although I do agree that more work should be done here to make these nested errors surface more gracefully. I will convert this to a roadmap item and then address it accordingly with the team soon.

Looking forward to seeing these cleaned up a bit for sure. Thanks for bringing this up!

**Comment by JarrodMFlesch** [comment]
We no longer use joi.
